android/sdk/src/main/java/org/jitsi/meet/sdk/analytics/AmplitudeModule.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright @ 2019-present 8x8, Inc.
+ * Copyright @ 2017-2018 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jitsi.meet.sdk.analytics;
+
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+
+import com.amplitude.api.Amplitude;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+/**
+ * Implements the react-native module for the Amplitude integration.
+ */
+public class AmplitudeModule
+        extends ReactContextBaseJavaModule {
+
+    public AmplitudeModule(ReactApplicationContext reactContext) {
+        super(reactContext);
+    }
+
+    /**
+     * Initializes the Amplitude SDK.
+     *
+     * @param instanceName The name of the Amplitude instance. Should
+     * be used only for multi-project logging.
+     * @param apiKey The API_KEY of the Amplitude project.
+     */
+    @ReactMethod
+    public void init(String instanceName, String apiKey) {
+        Amplitude.getInstance(instanceName).initialize(getCurrentActivity(), apiKey);
+    }
+
+    /**
+     * Sets the user properties for an Amplitude instance.
+     *
+     * @param instanceName The name of the Amplitude instance.
+     * @param userPropsString JSON string with user properties to be set.
+     */
+    @ReactMethod
+    public void setUserProperties(String instanceName, String userPropsString) {
+        JSONObject userProps = null;
+        try {
+            userProps = new JSONObject(userPropsString);
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+
+        Amplitude.getInstance(instanceName).setUserProperties(userProps);
+    }
+
+    /**
+     * Log an analytics event.
+     *
+     * @param instanceName The name of the Amplitude instance.
+     * @param eventType The event type.
+     * @param eventPropsString JSON string with the event properties.
+     */
+    @ReactMethod
+    public void logEvent(String instanceName, String eventType, String eventPropsString) {
```

Comment:
Same as above, have you measured if just passing an object is ok?

Reply:
I guess this comment and all other related to this question are dependant on https://github.com/jitsi/jitsi-meet/pull/3838#discussion_r251312576 conversation.